### PR TITLE
Feature/eclipse preferences

### PR DESCRIPTION
**Pull request by antonio-vazquez-araujo**
This is a possible implementation of the use of the Eclipse preference system. The projectConfigurators are much simpler and any change in the pom file makes the properties load automatically, without the user having to use "Maven->Update project". All the complexity of writing to property files is eliminated, and also, if the reference to the plugin is removed from the pom, the project properties are reset automatically .
I hope I haven't oversimplified the solution. 
Greetings.
Antonio.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@antonio-vazquez-araujo Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-io/spring-javaformat&pullRequestId=198)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-io/spring-javaformat&pullRequestId=198) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@antonio-vazquez-araujo Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-io/spring-javaformat&pullRequestId=198)!

File: spring-javaformat-eclipse/io.spring.javaformat.eclipse/src/io/spring/javaformat/eclipse/gradle/GradleProjectSettingsConfigurator.java [hunk 1]
```diff
@@ -16,117 +16,80 @@
...
-import org.eclipse.buildship.core.internal.workspace.InternalGradleWorkspace;
-import org.eclipse.buildship.core.internal.workspace.ModelProvider;
-import org.eclipse.core.resources.IProject;
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
-import org.gradle.tooling.CancellationTokenSource;
-import org.gradle.tooling.GradleConnector;
-import org.gradle.tooling.model.GradleTask;
-import org.gradle.tooling.model.eclipse.EclipseProject;
+import org.eclipse.m2e.core.lifecyclemapping.model.IPluginExecutionMetadata;
```
Comment (wilkinsona) [inline spring-javaformat-eclipse/io.spring.javaformat.eclipse/src/io/spring/javaformat/eclipse/gradle/GradleProjectSettingsConfigurator.java:25]: I'm surprised to see m2e types being imported into the Gradle settings configurator. Doesn't this need to hook into Buildship somehow?

Comment (antonio-vazquez-araujo) [reply spring-javaformat-eclipse/io.spring.javaformat.eclipse/src/io/spring/javaformat/eclipse/gradle/GradleProjectSettingsConfigurator.java]: Sorry, you are right. I've repaired it. 
Thanks.

**Comment by antonio-vazquez-araujo** [comment]
Hello  @wilkinsona. I would need to know if there is an intention to integrate this PR.
Thanks.

**Comment by wilkinsona** [comment]
Yes, i certainly hope so. We just need to find the time to do so and we’re rather busy with other things at the moment.

**Comment by philwebb** [comment]
Sorry for the delay in getting to this one. I managed to have a deeper look at the changes and I'm afraid I don't think we should apply it. It unfortunately drops two key features that we need:

* Custom settings in an `eclipse` or `.eclipse` folder are not found and applied.
* The copyright year from the `eclipse.properties` files is not applied.

I'm also a bit worried that the updated code could reset user configured settings in other projects.

It might be possible to rework `ProjectSettingsFiles.applyToProject` to use the `IEclipsePreferences`, but since the current solution doesn't seem to cause too many problems I'm tempted to keep things simple and just copy the bytes.

Thanks anyway for the PR.

**Pull request closed without merging**